te2rules/adapter.py

```diff
@@ -229,3 +233,121 @@ def _convert(self) -> DecisionTree:
 
         root_node = nodes[0]
         return root_node
+
+
+class XgboostXGBClassifierAdapter:
+    """
+    Class to convert xgboost.sklearn.XGBClassifier
+    into a te2rules.tree.RandomForest object.
+
+    Usage:
+    adapter = XgboostXGBClassifierAdapter(model, feature_names)
+    adapted_model = adapter.random_forest
+    """
+
+    def __init__(self, xgb_model: XGBClassifier, feature_names: List[str]):
+        self.xgb_model = xgb_model
+        self.feature_names = feature_names
+
+        # TODO: n0 and n1 are not accessible from the xgboost model itself, in this way bias term will be 0
+        self.n0 = 1
+        self.n1 = 1
+        self.bias = np.log(self.n1 / self.n0)
+
+        # Default learning rate 0.3
+        self.weight = self.xgb_model.get_params()["learning_rate"] or 0.3
+        self.activation = "sigmoid"
+
+        tree_ensemble = self._extract_tree_ensemble()
+        self.tree_ensemble = tree_ensemble
+
+        self.random_forest = self._convert()
+
+    def _extract_tree_ensemble(self) -> np.ndarray:
```

Comment:
This part of the code can be moved into the `_convert` method. Currently, the `_convert` method consumes the output of this method and returns it back as a different object.

Reply:
Yes, I would combine `_extract_tree_ensemble` and `_convert` methods into the same method named `_convert`